src/corppa/poetry_detection/annotation/recipe.py

```diff
@@ -234,12 +243,106 @@ def annotate_page_text(
+        components["before_db"] = lambda x: remove_image_data(
+            x, image_prefix=image_prefix
+        )
 
     return components
 
 
+def get_session_name(example: TaskType, suffix: Optional[str] = None) -> str:
+    """
+    Extract session name from task examplke. Session ids have the following form:
+        [session id] = [db id]-[session name]
+    Assumes that session names do not contain dashes.
+    """
+    session_id = example[SESSION_ID_ATTR]
+    session_name = session_id.rsplit("-", maxsplit=1)[-1]
+    if suffix is not None:
+        session_name = f"{session_name}-{suffix}"
+    return session_name
+
+
+def add_label_prefix(label: str, prefix: str) -> str:
+    return f"{prefix}: {label}"
+
+
+def remove_label_prefix(label: str) -> str:
+    return label.rsplit(": ", maxsplit=1)[-1]
+
+
+def add_session_prefix(
+    example: TaskType, session_sfx: Optional[str] = None
+) -> TaskType:
+    """
+    Add session name as prefix to text span labels
+
+    label --> "[session]: label"
+    ex. "POETRY" --> "alice: POETRY"
+
+    Calls: `get_sesssion_name` and `add_label_prefix`
+    """
+    session_name = get_session_name(example, suffix=session_sfx)
+    for span in example.get("spans", []):
+        span["label"] = add_label_prefix(span["label"], session_name)
+    return example
+
+
+def remove_session_prefix(example: TaskType) -> TaskType:
+    """
+    Remove added session prefix from text span labels
+
+    Calls: `remove_label_prefix`
+    """
+    for span in example.get("spans", []):
+        span["label"] = remove_label_prefix(span["label"])
+    return example
+
+
+def has_span_overlap(example: TaskType, strip_label_pfx: bool = True) -> bool:
+    """
+    Check if example has overlapping (text) span annotations
+
+    Calls: `remove_label_prefix` (optionally)
+    """
+    if "spans" not in example:
+        return False
+    label_coverage = {}  # label (str) --> coverage (intspan)
+    for span in example["spans"]:
+        # Get label
+        label = span["label"]
+        if strip_label_pfx:
+            label = remove_label_prefix(label)
+        # Represent span's coverage as an intspan
+        # Note: intspans are closed intervals, while span character ranges are half-open
+        span_coverage = intspan.from_range(span["start"], span["end"] - 1)
+        # Check if label's intspan is disjoint from
+        if label not in label_coverage:
+            label_coverage[label] = span_coverage
+        else:
+            if span_coverage.isdisjoint(label_coverage[label]):
+                # No overlap, accumulate into label's coverage
+                label_coverage[label] |= span_coverage
+            else:
+                # Overlap found
+                return True
+    return False
+
+
+def validate_review_answer(example: TaskType):
+    """
+    Validate review answer:
+    * Example not flagged (i.e. flagged field not set)
+    * Text spans (by label) must not overlap
```

Comment:
Your explanation in the comments was really clear & helpful - it would be good to include a brief version of it in the comment here. 

```suggestion
    * Text spans with the same label must not overlap (no duplicates or conflicting spans)
```

Reply:
How's the updated documentation look?

Reply:
great! thank you for clarifying